### PR TITLE
Feature/ 8 add section

### PR DESCRIPTION
**Pull request by Avareco**

### CODE REVIEW

File: src/components/MenuLeft.vue [hunk 1]
```diff
@@ -1,3 +1,96 @@
...
+    errorText.value = "Field can't be longer than 30 chars";
+    return false;
+  } else if (!name.trim()) {
+    errorText.value = "Field can't be empty";
+    return false;
+  } else if (
+    useTreeNodes().$state.items.find(
+      (item) => String(item.label).toLowerCase() === name.toLowerCase()
+    )
+  ) {
```
Comment (hollote) [inline src/components/MenuLeft.vue:20]: please format in a good way

File: src/functions/getItem.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { IItem } from "@/types/nodeTypes";
+
+export function getItem(item: IItem, path: string[]): IItem {
+  const arrIndex = Number(path.shift()) - 1;
+  if (path.length === 0) {
```
Comment (hollote) [inline src/functions/getItem.ts:5]: if (!path.length) {

Comment (hollote) [reply src/functions/getItem.ts:5]: and if you're using return no need to use else block

File: src/functions/nextNodeKey.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  let firstPartOfKey = path.slice(0, path.length - 1).join("_");
+  let lastNumbersOfKey = currentNodeKey;
+
+  if (currentNodeKey.includes("_")) {
+    lastNumbersOfKey = currentNodeKey.slice(
+      currentNodeKey.lastIndexOf("_") + 1
+    );
+  }
+
+  if (firstPartOfKey) firstPartOfKey = firstPartOfKey + "_";
```
Comment (hollote) [inline src/functions/nextNodeKey.ts:12]: please use brackets with if(...) {
    ...
}

File: src/components/MenuLeft.vue (hunk 1 above)
Comment (Avareco) [reply src/components/MenuLeft.vue:20]: fixed

File: src/functions/getItem.ts (hunk 2 above)
Comment (Avareco) [reply src/functions/getItem.ts:5]: fixed

File: src/functions/nextNodeKey.ts (hunk 3 above)
Comment (Avareco) [reply src/functions/nextNodeKey.ts:12]: fixed

**Comment by Avareco** [comment]
Hi, @hollote. Screenshots:
image
image
image
image

**Comment by Avareco** [comment]
image
image

File: src/functions/createBreadcrumbs.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { IItem } from "@/types/nodeTypes";
+
+export function createBreadcrumbs(item: IItem, path: string[], breadcrumbs) {
+  const arrIndex = Number(path.shift()) - 1;
+  breadcrumbs = [...breadcrumbs, item[arrIndex]];
+  if (path.length === 0) {
```
Comment (hollote) [inline src/functions/createBreadcrumbs.ts:6]: same if condition

Comment (Avareco) [reply src/functions/createBreadcrumbs.ts]: fixed, added types

File: src/components/MenuLeft.vue (hunk 1 above)
Comment (Avareco) [reply src/components/MenuLeft.vue:24]: fixed

**Pull request merged**